### PR TITLE
Adjust recruitment window position on castle well and kingdom overview

### PR DESCRIPTION
**Pull request by Kazhuu**
Solves https://github.com/ihhub/fheroes2/issues/4749. The recruitment windows is lowered 2 pixels on castle well view and in kingdom overview centered over top part of the window.

### CODE REVIEW

**Comment by Branikolog** [comment]
Sorry for the late response.
What about adjusting creature info window position too?
image
The same as recruiting window was made.

**Comment by Kazhuu** [comment]
> Sorry for the late response. What about adjusting creature info window position too? image The same as recruiting window was made.

Good catch! I now moved these as well Hopefully now all windows are centered like they should.

**Comment by LeHerosInconnu** [comment]
Hello @Kazhuu and Branikolog,

> Hi, @Kazhuu ! Sorry for the late response.
> 
> > Hopefully now all windows are centered like they should.
> 
> Unluckily, windows in castle and hero info are shifted now too:
> image

For this one, I think it would be better to center the window in the lower part of the screen so that it is possible to see clearly the numbers of the hero's primary skills:
Hero screen cadre rouge 001

> image

For this one, I would leave it as it is, because it allows to see the creatures and their numbers below the window.

**Comment by ihhub** [comment]
Hi @LeHerosInconnu , let's focus in this pull request only on changes which are related to the original issue. It is good to keep track on them.

**Comment by Kazhuu** [comment]
I now did the changes so that the only kingdom overview troop info and recruit windows are moved, not those shown on castle and hero windows. I also adjusted the troop info window on kingdom overview when heroes and their troops are shown. This was not adjusted before.
image

**Comment by ihhub** [comment]
Hi @Branikolog , please review this pull request.

**Review by Branikolog — APPROVED** [review]
I haven't found any issues here. 👍

File: src/fheroes2/army/army_bar.h [hunk 1]
```diff
@@ -81,6 +83,7 @@ class ArmyBar : public Interface::ItemsActionBar<ArmyTroop>
     bool read_only;
     bool can_change;
     std::string msg;
+    int32_t troopWindowOffsetY;
```
Comment (ihhub) [inline src/fheroes2/army/army_bar.h:86]: Please name newly added member of the class with an underscore character: **_troopWindowOffsetY**.

File: src/fheroes2/dialog/dialog.h [hunk 2]
```diff
@@ -100,7 +100,7 @@ namespace Dialog
     bool InputString( const std::string & header, std::string & result, const std::string & title = std::string(), const size_t charLimit = 0 );
     Troop RecruitMonster( const Monster & monster0, uint32_t available, const bool allowDowngradedMonster, const int32_t windowOffsetY );
     void DwellingInfo( const Monster &, uint32_t available );
-    int ArmyInfo( const Troop & troop, int flags, bool isReflected = false );
+    int ArmyInfo( const Troop & troop, int flags, bool isReflected = false, int32_t windowOffsetY = 0 );
```
Comment (ihhub) [inline src/fheroes2/dialog/dialog.h:103]: Please add **const** for `windowOffsetY`.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @Kazhuu , I put several comments here. Could you please take a look when you have time?

**Comment by Kazhuu** [comment]
@ihhub I did the changes requested :+1:

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@Kazhuu , thank you so much for this improvement!

**Pull request merged**